src/libs/ajax/DataRepo.ts

```diff
@@ -103,6 +104,7 @@ export interface DataRepoContract {
   dataset: (datasetId: string) => {
     details: (include?: DatasetInclude[]) => Promise<DatasetModel>;
     roles: () => Promise<string[]>;
+    createDatasetRequest(_request: DatasetAccessRequestAPI): Promise<DatasetAccessRequestAPI>;
```

Comment:
The terra-ui code uses a `_` prefix to indicate that a parameter isn't used to avoid an eslint warning. In this case this is the type signature so there's no function body to use the parameter so you don't need the `_` form to avoid the warning.
```suggestion
    createDatasetRequest(request: DatasetAccessRequestAPI): Promise<DatasetAccessRequestAPI>;
```